converters/toZigbee.js

```diff
@@ -190,6 +190,294 @@ const converters = {
+                return {
+                    cid: cid,
+                    cmd: 'read',
+                    cmdType: 'foundation',
+                    zclData: [{attrId: zclId.attr(cid, attrId).value}],
+                    cfg: cfg.default,
+                };
+            }
+        },
+    },
+    thermostat_weeklySchedule: { // not tested
+        key: 'weeklySchedule',
+        convert: (key, value, message, type) => {
+            const cid = 'hvacThermostat';
+            const attrId = 'weeklySchedule';
+            if (type === 'set') {
+                return {
+                    cid: cid,
+                    cmd: 'setWeeklySchedule',
+                    cmdType: 'functional',
+                    zclData: {
+                        dataType: 0x29, // dataType
+                        attrData: Math.round(value) * 100,
+                        numoftrans: value.numoftrans,
+                        dayofweek: value.dayofweek,
+                        mode: value.mode,
+                        thermoseqmode: value.thermoseqmode,
+                    },
+                    cfg: cfg.default,
+                };
+            } else if (type === 'get') { // MAC transaction expired.
+                return {
+                    cid: cid,
+                    cmd: 'getWeeklySchedule',
+                    cmdType: 'functional',
+                    zclData: [
+                        {attrId: zclId.attr(cid, attrId).value},
+                        // {daystoreturn: value.daystoreturn},
+                        // {modetoreturn: value.modetoreturn},
+                    ],
+                    cfg: cfg.default,
+                };
+            }
+        },
+    },
+    thermostat_clearWeeklySchedule: { // not tested
+        key: 'clearWeeklySchedule',
+        attr: [],
+        convert: (key, value, message, type) => {
+            return {
+                cid: 'hvacThermostat',
+                cmd: 'clearWeeklySchedule',
+                type: 'functional',
+                zclData: {},
+            };
+        },
+    },
+    thermostat_getRelayStatusLog: { // not tested
+        key: 'getRelayStatusLog',
+        attr: [],
+        convert: (key, value, message, type) => {
+            return {
+                cid: 'hvacThermostat',
+                cmd: 'getRelayStatusLog',
+                type: 'functional',
+                zclData: {},
+            };
+        },
+    },
+    thermostat_getWeeklyScheduleRsp: { // not tested
+        key: 'getWeeklyScheduleRsp',
+        attr: [],
+        convert: (key, value, message, type) => {
+            return {
+                cid: 'hvacThermostat',
+                cmd: 'getWeeklyScheduleRsp',
+                type: 'functional',
+                zclData: {
+                    numoftrans: value.numoftrans,
+                    dayofweek: value.dayofweek,
+                    mode: value.mode,
+                    thermoseqmode: value.thermoseqmode,
```

Comment:
Please change to: number_of_transitions (I guess)?, day_of_week, thermostat_sequence_mode (?)